integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/projection/IdentifierSearchProjectionSingleValuedIT.java

```diff
@@ -0,0 +1,205 @@
+				.toQuery();
+
+		List<Object> result = query.fetchHits( 30 );
+		assertThat( result ).containsExactlyInAnyOrder( (Object[]) ids );
+	}
+
+	@Test
+	public void duplicated() {
+		StubMappingScope scope = index.createScope();
+
+		SearchQuery<List<?>> query = scope.query()
+				.select( f -> f.composite( f.id( String.class ), f.id( String.class ) ) )
+				.where( f -> f.matchAll() )
+				.toQuery();
+
+		List<List<?>> result = query.fetchHits( 30 );
+		assertThat( result ).containsExactlyInAnyOrder( duplicates );
+	}
+
+	@Test
+	public void validSuperClass() {
+		StubMappingScope scope = index.createScope();
+		SearchQuery<CharSequence> query = scope.query()
+				.select( f -> f.id( CharSequence.class ) )
+				.where( f -> f.matchAll() )
+				.toQuery();
+
+		List<CharSequence> result = query.fetchHits( 30 );
+		assertThat( result ).containsExactlyInAnyOrder( ids );
+	}
+
+	@Test
+	public void invalidProjectionType() {
+		StubMappingScope scope = index.createScope();
+
+		assertThatThrownBy( () -> scope.projection().id( Integer.class ) )
+				.isInstanceOf( SearchException.class )
+				.hasMessageContainingAll(
+						"The required identifier type 'java.lang.Integer' does not match the actual identifier type 'java.lang.String'",
+						"the required identifier must be a superclass of the actual identifier"
+				);
+	}
+
+	@Test
+	public void multiIndex_withCompatibleIndex() {
+		StubMappingScope scope = index.createScope( compatibleIndex );
+		SearchQuery<String> query = scope.query()
+				.select( f -> f.id( String.class ) )
+				.where( f -> f.matchAll() )
+				.toQuery();
+
+		List<String> result = query.fetchHits( 30 );
+		assertThat( result ).containsExactlyInAnyOrder( ids );
+	}
+
+	@Test
+	public void multiIndex_withIncompatibleIndex() {
+		StubMappingScope scope = index.createScope( incompatibleIndex );
+
+		assertThatThrownBy( () -> scope.projection().id() )
+				.isInstanceOf( SearchException.class )
+				.hasMessageContainingAll(
+						"Inconsistent configuration for the identifier in a search query across multiple indexes",
+						"converter differs"
+				);
+	}
+
+	@Test
+	public void nullClass() {
+		StubMappingScope scope = index.createScope();
+
+		assertThatThrownBy( () -> scope.projection().id( (Class<?>) null ) )
+				.isInstanceOf( IllegalArgumentException.class )
+				.hasMessageContainingAll(
+						"must not be null", "identifierType"
+				);
+	}
+
+	private void initValues() {
+		for ( int i = 0; i < 7; i++ ) {
+			int suffix = i + 1;
+			ids[i] = "ID-" + suffix;
+			names[i] = "my-name-" + suffix;
+
+			ArrayList<String> duplicated = new ArrayList<>( 2 );
+			duplicated.add( ids[i] );
+			duplicated.add( ids[i] );
+			duplicates[i] = duplicated;
+		}
+	}
+
+	private void initData() {
+		BulkIndexer indexer = index.bulkIndexer();
+		for ( int i = 0; i < 7; i++ ) {
+			int finalI = i;
+			indexer.add( ids[i], f -> f.addValue( index.binding().name, names[finalI] ) );
+		}
+
+		indexer.join();
```

Comment:
FYI you can also do this:

```suggestion
		index.bulkIndexer()
				.add( 7, i -> documentProvider(
						ids[i],
						document -> document.addValue( index.binding().name, names[i] )
				) )
				.join();
```